### ISSUE TITLE
ListAutoPaging requires async capabilities?

### ISSUE THREAD

**Issue by mexner** [stripe/stripe-dotnet#2223]
Using webforms (I know.. I know...), our pages are setup to post back to the server and we have not enabled any async capabilities on the page. The page's code-behind is calling 

```
var invoiceService = new InvoiceService(_client);
var options = new InvoiceListOptions()
{
       Limit = 50,
       Customer = customerId,
};

var invoices = invoiceService.ListAutoPaging(options).ToList();
return invoices;
```

Using Fidler I can see that the service is contacting Stripe and returning 50 records. 
The system just spins and does nothing until the thread expires if there are more than 50 records, despite the returned JSON property "has_more" = true. 

I believe this is because the underlying/base "ListAutoPaging" feature has async dependencies, as found here,
https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Services/_base/Service.cs#L260

Thus, the service works in .net web-forms, but never completes because the async services stall out when web-forms are not setup to support async. 

I will work around this with our own do/while loop, but wanted to raise this for others and verify I'm not missing anything?

Using Visual Studio Version 16.7.3
.NET v4.7.2, C#

**Comment by MarkA1** [comment]
@mexner, We reported and worked through what is likely the same issue but on InvoiceService.ListLineItemsAutoPaging, and finding the cause of this with Stripe support a couple of weeks ago, though still no report of a fix or work in progress.

We're using .NET Framework v4.7.2, C# with MVC and discovered the issue after upgrading to Stripe .Net v39.2.0 from a much older version.  

It looks like the issue started at v35.11.0 with the introduction of feature #1973 Support for asynchronous autopagination.  The issue occurs in AsyncUtils here on line 35: https://github.com/stripe/stripe-dotnet/blob/master/src/Stripe.net/Infrastructure/AsyncUtils.cs#L35

I was able to load the source locally, step into the Stripe code and reliably reproduce the error at the above line.

We're hoping to see a fix soon as well.
